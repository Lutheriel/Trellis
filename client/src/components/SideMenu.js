import React, { useState } from 'react'
import { Paper, makeStyles , fade} from '@material-ui/core'
import AccountTreeIcon from '@material-ui/icons/AccountTree'
import MoreHorizIcon from '@material-ui/icons/MoreHoriz'
import InputBase from '@material-ui/core/InputBase'
import SearchIcon from '@material-ui/icons/Search'
import { useDispatch, useSelector } from 'react-redux'
import DeleteSweepIcon from '@material-ui/icons/DeleteSweep'
import { Link } from 'react-router-dom'
import AddItem from './AddItem'
import Activities from './Activities'
import Hr from './Hr'
import MenuHeader from './MenuHeader'
import Background from './Background'
import { deleteBoardById } from '../actions/actionCreators/boardActions'
const useStyles = makeStyles((theme) => ({
  container: {
    backgroundColor: '#F4F5F7',
    width: '321px',
    float: 'right',
    height: '100vh',
    right: (props) =>
      props.showMenu === false ? theme.spacing(-100) : theme.spacing(0),
    top: theme.spacing(5.63),
    borderRadius: theme.spacing(0),
    position: 'fixed',
    padding: theme.spacing(1),
    wordWrap: 'break-word',
    zIndex: '1200',
    transition: 'right 0.7s ease-out',
  },
  scroll: {
    overflow: 'auto',
    height: '95vh',
  },
  menu: {
    top: theme.spacing(5.63),
    position: 'fixed',
    right: theme.spacing(0),
    marginTop: theme.spacing(0.5),
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginRight: theme.spacing(2),
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing(3),
      width: 'auto',
    },
  },
  searchIcon: {
    padding: theme.spacing(0, 2),
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 0),
    // vertical padding + font size from searchIcon
    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: '20ch',
    },
  }
}))
export default function SideMenu({ setBackground, board }) {
  const [showMenu, setShowMenu] = useState(false)
  const [showBackground, setShowBackground] = useState(false)
  const classes = useStyles({ showMenu })
  const dispatch = useDispatch()
  const { token } = useSelector((state) => state.user)
  return (
    <>
      <div className={classes.menu}>
        <AddItem
          btnText="Show Menu"
          handleClick={() => setShowMenu(true)}
          icon={<MoreHorizIcon />}
          type="menu"
          width="120px"
          color="white"
        />
      </div>
      {!showBackground && (
        <Paper className={classes.container} elevation={1} variant="outlined">
          <MenuHeader
            text="Menu"
            closeHandler={() => setShowMenu(false)}
            type="menu"
          />
          <Hr />
          <div className={classes.search}>
            <div className={classes.searchIcon}>
              <SearchIcon />
            </div>
            <InputBase
              placeholder="Searchâ€¦"
              classes={{
                root: classes.inputRoot,
                input: classes.inputInput,
              }}
              inputProps={{ 'aria-label': 'search' }}
            />
          </div>
          <Hr />
          <AddItem
            btnText="Change Background"
            handleClick={() => setShowBackground(true)}
            type="background"
            width="310px"
            icon={
              <span
                style={{
                  marginRight: '13px',
                  backgroundColor: `${board.color}`,
                  backgroundImage: `url(${board.url})`,
                  backgroundSize: 'cover',
                  backgroundRepeat: 'no-repeat',
                  width: '20px',
                  height: '20px',
                  borderRadius: '2px',
                }}
              ></span>
            }
          />
          <Link to="/" style={{ textDecoration: 'none' }}>
            <AddItem
              btnText="Delete Board"
              handleClick={() => {
                dispatch(deleteBoardById(board.id, token))
              }}
              type="background"
              width="310px"
              icon={<DeleteSweepIcon style={{ marginRight: '10px' }} />}
            />
          </Link>
          <div style={{ display: 'flex', marginTop: '20px' }}>
            <AccountTreeIcon
              fontSize="small"
              style={{ paddingLeft: '20px', color: '#172B4D', opacity: '0.8' }}
            />
            <div
              style={{
                paddingLeft: '10px',
                paddingBottom: '10px',
                fontWeight: 'bold',
                fontSize: '15px',
                color: '#172B4D',
                opacity: '0.9',
              }}
            >
              Activity
            </div>
          </div>
          <div className={classes.scroll}>
            <Activities />
          </div>
        </Paper>
      )}
      <div>
        {showBackground && (
          <Background
            backHandler={() => setShowBackground(false)}
            closeHandler={() => {
              setShowMenu(false)
              setShowBackground(false)
            }}
            setColorBackground={setBackground}
          />
        )}
      </div>
    </>
  )
}
