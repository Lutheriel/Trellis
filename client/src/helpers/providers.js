import React from 'react'
import { Provider } from 'react-redux'
import store from '../store/configureStore'
import { DragDropContext, Droppable } from 'react-beautiful-dnd'

export const TestProvider = ({ children }) => {
  return (
    <Provider store={store}>
      <DragDropContext onDragEnd={() => {}}>{children}</DragDropContext>
    </Provider>
  )
}
