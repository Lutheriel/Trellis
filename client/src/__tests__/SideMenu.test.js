import React from 'react';
import { render } from '@testing-library/react';
import SideMenu from '../components/SideMenu';
import renderer from 'react-test-renderer';
import { TestProvider } from '../helpers/providers'


it('should display correctly SideMenu', async () => {
    const { getByText, getByTestId } = render(
      <TestProvider>
        <SideMenu
        setBackground={() => {}}
        board={{title: 'title' }}
        setSearch={() => {}}
        search={''}
        isResultEmpty={true}
      />
      </TestProvider>
    );
  
    const menu = getByText('Menu');
    expect(window.getComputedStyle(menu).display).toEqual('block');

  });
  it('should check snapshot for SideMenu', () => {
    const tree = renderer
      .create(
        <TestProvider>
        <SideMenu
        setBackground={() => {}}
        board={{title: 'title' }}
        setSearch={() => {}}
        search={''}
        isResultEmpty={true}
      />
      </TestProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

