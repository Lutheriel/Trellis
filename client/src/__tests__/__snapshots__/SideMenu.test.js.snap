// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should check snapshot for SideMenu 1`] = `
Array [
  <div
    className="makeStyles-menu-26"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-add-32 makeStyles-add-34 makeStyles-width-33 makeStyles-width-35"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
         
        Show Menu
      </span>
    </button>
  </div>,
  <div
    className="MuiPaper-root makeStyles-container-24 makeStyles-container-31 MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="makeStyles-title-36 makeStyles-title-38"
      >
        Menu
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-close-37"
        data-testid="MenuHeader-close-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
      </button>
    </div>
    <hr
      className="makeStyles-hr-39"
    />
    <div
      className="makeStyles-search-27"
    >
      <div
        className="makeStyles-searchIcon-28"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <div
        className="MuiInputBase-root makeStyles-inputRoot-29"
        onClick={[Function]}
      >
        <input
          aria-label="search"
          className="MuiInputBase-input makeStyles-inputInput-30"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search by keyword"
          type="text"
          value=""
        />
      </div>
    </div>
    
    <hr
      className="makeStyles-hr-39"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-add-32 makeStyles-add-40 makeStyles-width-33 makeStyles-width-41"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          style={
            Object {
              "backgroundColor": "undefined",
              "backgroundImage": "url(undefined)",
              "backgroundRepeat": "no-repeat",
              "backgroundSize": "cover",
              "borderRadius": "2px",
              "height": "20px",
              "marginRight": "13px",
              "width": "20px",
            }
          }
        />
         
        Change Background
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-add-32 makeStyles-add-42 makeStyles-width-33 makeStyles-width-43"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "marginRight": "10px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M15 16h4v2h-4zm0-8h7v2h-7zm0 4h6v2h-6zM3 18c0 1.1.9 2 2 2h6c1.1 0 2-.9 2-2V8H3v10zM14 5h-3l-1-1H6L5 5H2v2h12z"
          />
        </svg>
         
        Delete Board
      </span>
    </button>
    <div
      style={
        Object {
          "display": "flex",
          "marginTop": "20px",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        style={
          Object {
            "color": "#172B4D",
            "opacity": "0.8",
            "paddingLeft": "20px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3z"
        />
      </svg>
      <div
        style={
          Object {
            "color": "#172B4D",
            "fontSize": "15px",
            "fontWeight": "bold",
            "opacity": "0.9",
            "paddingBottom": "10px",
            "paddingLeft": "10px",
          }
        }
      >
        Activity
      </div>
    </div>
    <div
      className="makeStyles-scroll-25"
    >
      <div
        className="makeStyles-wrapper-44"
      />
    </div>
  </div>,
  <div />,
]
`;
